db.go

```diff
@@ -31,6 +31,14 @@ const (
 	GetIdentitiesFlagDeleted GetIdentitiesFlag = 1 << (iota - 1)
 )
 
+type AuthCheck int
+
+const (
+	AuthCheckNone             AuthCheck = 0
+	AuthCheckNonPasswordReuse AuthCheck = 1 << (iota - 1)
+	AuthCheckPasswordExpired
+)
```

Comment:
The idea was to use bit-wise comparison instead of standard int's. You can see this by adding an extra item to the enum.

```
const (
	AuthCheckNone             AuthCheck = 0
	AuthCheckNonPasswordReuse AuthCheck = 1<<(iota -1)
	AuthCheckPasswordExpired
        AuthCheckPasswordNext
)
```
AuthCheckPasswordNext will be different in these scenarios.